taskt/UI/Forms/ScriptBuilder Forms/frmScriptBuilder.cs

```diff
@@ -0,0 +1,533 @@
+            {
+                TreeNode newGroup = new TreeNode(cmd.Key);
+
+                foreach (var subcmd in cmd)
+                {
+                    TreeNode subNode = new TreeNode(subcmd.ShortName);
+
+                    if (!subcmd.Command.CustomRendering)
+                    {
+                        subNode.ForeColor = Color.Red;
+                    }
+                    newGroup.Nodes.Add(subNode);
+                }
+
+                tvCommands.Nodes.Add(newGroup);
+            }
+
+            tvCommands.Sort();
+            //tvCommands.ImageList = uiImages;
+
+            //start attended mode if selected
+            if (_appSettings.ClientSettings.StartupMode == "Attended Task Mode")
+            {
+                WindowState = FormWindowState.Minimized;
+                var frmAttended = new frmAttendedMode();
+                frmAttended.Show();
+            }
+
+        }
+
+        private void frmScriptBuilder_FormClosing(object sender, FormClosingEventArgs e)
+        {
+            DialogResult result = CheckForUnsavedScripts();
+            if (result == DialogResult.Cancel)
+                e.Cancel = true;
+        }
+
+        private void GenerateRecentFiles()
+        {
+            flwRecentFiles.Controls.Clear();
+
+            var scriptPath = Folders.GetFolder(Folders.FolderType.ScriptsFolder);
+
+            if (!Directory.Exists(scriptPath))
+            {
+                lblRecentFiles.Text = "Script Folder does not exist";
+                lblFilesMissing.Text = "Directory Not Found: " + scriptPath;
+                lblRecentFiles.ForeColor = Color.White;
+                lblFilesMissing.ForeColor = Color.White;
+                lblFilesMissing.Show();
+                flwRecentFiles.Hide();
+                return;
+            }
+
+            var directory = new DirectoryInfo(scriptPath);
+            var recentFiles = directory.GetFiles()
+                                       .OrderByDescending(file => file.LastWriteTime)
+                                       .Select(f => f.Name);
+
+            if (recentFiles.Count() == 0)
+            {
+                //Label noFilesLabel = new Label();
+                //noFilesLabel.Text = "No Recent Files Found";
+                //noFilesLabel.AutoSize = true;
+                //noFilesLabel.ForeColor = Color.SteelBlue;
+                //noFilesLabel.Font = lnkGitIssue.Font;
+                //noFilesLabel.Margin = new Padding(0, 0, 0, 0);
+                //flwRecentFiles.Controls.Add(noFilesLabel);
+                lblRecentFiles.Text = "No Recent Files Found";
+                lblRecentFiles.ForeColor = Color.White;
+                lblFilesMissing.ForeColor = Color.White;
+                lblFilesMissing.Show();
+                flwRecentFiles.Hide();
+            }
+            else
+            {
+                foreach (var fil in recentFiles)
+                {
+                    if (flwRecentFiles.Controls.Count == 7)
```

Comment:
Any particular motivation for the hard coded 7?

Reply:
I believe it's because on the panel that loads when you create a new file, links to the 7 most recently modified files in My Script Folder are generated

Reply:
I've been debating whether to remove that now that we have the project structure. Or change it to links to the Mains of the most recent projects instead